packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -780,19 +800,30 @@ export class RunningSummarizer implements IDisposable {
 			}
 		} while (!done);
 
+		// If summarize attempt did not fail, emit "summarize" event and return. A failed attempt may be retried below.
+		if (result !== "failure") {
+			this.emit("summarize", { result, currentAttempt, maxAttempts });
+			return;
+		}
+
 		// If summarization wasn't successful above and the failure contains "retryAfterSeconds", perform one last
 		// attempt. This gives a chance to the runtime to perform additional steps in the last attempt.
-		if (!success && retryAfterSeconds !== undefined) {
+		if (retryAfterSeconds !== undefined) {
```

Comment:
@tyler-cai-microsoft  - Do you think we should check cancellation token here first before doing the final attempt?

Reply:
I don't see why not